ADAL/src/cache/ADTokenCacheAccessor.m

```diff
@@ -63,25 +64,45 @@ - (id)initWithDataSource:(id<ADTokenCacheDataSource>)dataSource
     return _dataSource;
 }
 
-- (ADTokenCacheItem *)getItemForUser:(ADUserIdentifier *)identifier
+- (ADTokenCacheItem *)getItemForUser:(NSString *)userId
                             resource:(NSString *)resource
                             clientId:(NSString *)clientId
                              context:(id<ADRequestContext>)context
                                error:(ADAuthenticationError * __autoreleasing *)error
 {
-    ADTokenCacheKey* key = [ADTokenCacheKey keyWithAuthority:_authority
-                                                    resource:resource
-                                                    clientId:clientId
-                                                       error:error];
-    if (!key)
+    NSArray<NSURL *> *aliases = [[ADAuthorityValidation sharedInstance] cacheAliasesForAuthority:[NSURL URLWithString:_authority]];
+    for (NSURL *alias in aliases)
     {
-        return nil;
+        ADTokenCacheKey* key = [ADTokenCacheKey keyWithAuthority:[alias absoluteString]
+                                                        resource:resource
+                                                        clientId:clientId
+                                                           error:error];
+        if (!key)
+        {
+            return nil;
```

Comment:
Just theoretically: if we fail to create a key with one given alias, maybe we should try with the next one instead of returning early?

Reply:
If we fail to return a key then we have some bigger problem that I'd prefer not to mask seeing how the key is little more then a bunch of strings glued together.